### PR TITLE
[redesign] initial Table component

### PR DESCRIPTION
**Pull request by Simek**
# Why

Refs https://linear.app/expo/issue/ENG-1705

# How

This PR add the initial `Table` component and related ones like `Row` and `Cell`.

The code or structure of those component might change, I also had to add overwrites for the current global table styles, which could be removed when global table CSS will be adapted to the new design (which is used for pure Markdown pages).

# Test Plan

Test table has been added to `test-design` page.

Screenshot 2021-08-06 at 14 19 46

### CODE REVIEW

File: docs/ui/components/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+type TableProps = {
+  headers: string[];
+  children: React.ReactNode;
+};
+
+export const Table: React.FC<TableProps> = ({ children, headers }) => (
+  <div css={tableWrapperStyle}>
+    <table css={tableStyle}>
+      {headers && headers.length ? (
```
Comment (byCedric) [inline docs/ui/components/Table.tsx:13]: I'm not sure if adding a `headers = []` default value would help, but also wanted to add that as alternative for this check 😄

File: docs/ui/components/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { css } from '@emotion/react';
+import { theme, borderRadius } from '@expo/styleguide';
+import React from 'react';
+
+type TableProps = {
+  headers: string[];
+  children: React.ReactNode;
+};
+
+export const Table: React.FC<TableProps> = ({ children, headers }) => (
```
Comment (byCedric) [inline docs/ui/components/Table.tsx:10]: I think the `React.FC` is deprecated and might be better to move off from these types?

- https://stackoverflow.com/questions/53885993/react-16-7-react-sfc-is-now-deprecated (SO overflow about this)
- https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/ (detailed explanation)

**Review by byCedric — APPROVED** [review]
Looks good, great work! ❤️

File: docs/ui/components/Table.tsx (hunk 2 above)
Comment (Simek) [reply docs/ui/components/Table.tsx:10]: Thank you fo the hint about that, I did know that! 🙂

There is an origin PR which deprecated FC or FunctionalComponent types:
* https://github.com/facebook/create-react-app/pull/8177

Also it looks like in this thread and in the article you have mentioned there is a suggested refactor which looks quite nice, so I will apply that.

**Comment by Simek** [comment]
Hope I did not go too far with files granulation and `TextAlign` isn't an over-complication, but I'm fine with changing it or even reverting the whole commit 😄

**Review by byCedric — APPROVED** [review]

**Pull request merged**